### PR TITLE
[Spark] Fix the inaccurate issue of obtaining COLUMN_SIZE in the decimal field jdbc of spark engine

### PR DESCRIPTION
**Pull request by waywtdcc**
# :mag: Description
## Issue References 🔗

This pull request fixes #

## Describe Your Solution 🔧

Fix the inaccuracy in getting COLUMN_SIZE from spark engine's decimal field jdbc. The current jdbc's get column size gets the decimal field as default size, which is inaccurate; if it is decimal(20,3), the obtained column size is 16; the actual is 20.

## Types of changes :bookmark:

- [X] Bugfix (non-breaking change which fixes an issue)

#### Related Unit Tests

---

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala [hunk 1]
```diff
@@ -142,7 +142,12 @@ object SchemaHelper {
           .contains(dt.getClass.getSimpleName) => Some(dt.defaultSize)
     case dt @ (BooleanType | _: NumericType | DateType | TimestampType |
         CalendarIntervalType | NullType) =>
-      Some(dt.defaultSize)
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala:145]: We can add another case branch like `case dt: DecimalType =>`

**Comment by yaooqinn** [comment]
Nice catch! Hi @waywtdcc , can we restore the checklist from PR desc?

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala [hunk 2]
```diff
@@ -140,6 +140,8 @@ object SchemaHelper {
     case dt
         if Array(TIMESTAMP_NTZ, DAY_TIME_INTERVAL, YEAR_MONTH_INTERVAL)
           .contains(dt.getClass.getSimpleName) => Some(dt.defaultSize)
+    case dt @ (_: DecimalType) =>
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala:143]: why do we need @ here？

**Review by yaooqinn — CHANGES_REQUESTED** [review]
Hi, please restore the PR description

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala (hunk 2 above)
Comment (waywtdcc) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala:143]: case dt: DecimalType =>
      Some(dt.precision) ？

Comment (yaooqinn) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala:143]: yeah

Comment (yaooqinn) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala:143]: @waywtdcc can you fix this?

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
Merged to master, 1.8, 1.7

**Pull request closed without merging**